pkg/sql/catalog/lease/lease.go

```diff
@@ -1130,28 +1130,49 @@ func (m *Manager) RefreshLeases(ctx context.Context, s *stop.Stopper, db *kv.DB)
 				}
 
 				if evFunc := m.testingKnobs.TestingDescriptorUpdateEvent; evFunc != nil {
-					if err := evFunc(desc); err != nil {
+					if err := evFunc(desc.DescriptorProto()); err != nil {
 						log.Infof(ctx, "skipping update of %v due to knob: %v",
 							desc, err)
 						continue
 					}
 				}
 
-				id, version, name, state, err := descpb.GetDescriptorMetadata(desc)
-				if err != nil {
-					log.Fatalf(ctx, "invalid descriptor %v: %v", desc, err)
-				}
-				dropped := state == descpb.DescriptorState_DROP
+				dropped := desc.Dropped()
 				// Try to refresh the lease to one >= this version.
 				log.VEventf(ctx, 2, "purging old version of descriptor %d@%d (dropped %v)",
-					id, version, dropped)
-				if err := purgeOldVersions(ctx, db, id, dropped, version, m); err != nil {
-					log.Warningf(ctx, "error purging leases for descriptor %d(%s): %s",
-						id, name, err)
+					desc.GetID(), desc.GetVersion(), dropped)
+				purge := func(ctx context.Context) {
+					if err := purgeOldVersions(ctx, db, desc.GetID(), dropped, desc.GetVersion(), m); err != nil {
+						log.Warningf(ctx, "error purging leases for descriptor %d(%s): %s",
+							desc.GetID(), desc.GetName(), err)
+					}
+				}
+				// New descriptors may appear in the future if the descriptor table is
```

Comment:
Why the table's modification time can be in the future? It's because the clock difference between regions?